PyStemmusScope/forcing_io.py

```diff
@@ -37,60 +43,147 @@ def read_forcing_data(forcing_file, start_time, end_time):
-    data['elevation'] = ds_forcing['elevation'].values
-    data['IGBP_veg_long'] = ds_forcing['IGBP_veg_long'].values
-    data['reference_height'] = ds_forcing['reference_height'].values
-    data['canopy_height'] = ds_forcing['canopy_height'].values
+    time_delta = (
+        ds_forcing.time.values[1] - ds_forcing.time.values[0]
+    ) / np.timedelta64(1, "s")
+    data["DELT"] = time_delta.astype(float)
+    data["total_timesteps"] = ds_forcing.time.size
+
+    data["latitude"] = ds_forcing["latitude"].values
+    data["longitude"] = ds_forcing["longitude"].values
+    data["elevation"] = ds_forcing["elevation"].values
+    data["IGBP_veg_long"] = ds_forcing["IGBP_veg_long"].values
+    data["reference_height"] = ds_forcing["reference_height"].values
+    data["canopy_height"] = ds_forcing["canopy_height"].values
 
     # these are needed by save.py
-    data['time'] = ds_forcing["time"]
-    data['Qair'] = ds_forcing['Qair']
+    data["time"] = ds_forcing["time"]
+    data["Qair"] = ds_forcing["Qair"]
+
+    return data
+
+
+def read_forcing_data_global(
+    global_data_dir: Path,
+    lat: float,
+    lon: float,
+    start_time: np.datetime64,
+    end_time: np.datetime64,
+) -> Dict:
+    """Read forcing data for a certain location, based on global datasets.
+
+    Args:
+        global_data_dir: Path to the directory containing the global datasets.
+        lat: Latitude of the site of interest.
+        lon: Longitude of the site of interest.
+        start_time: Start time of the model run.
+        end_time: End time of the model run.
+
+    Returns:
+        Dictionary containing the forcing data.
+    """
+
+    files_cams = list((global_data_dir / "co2").glob("*.nc"))
+    file_canopy_height = (
+        global_data_dir / "canopy_height" / gds.get_filename_canopy_height(lat, lon)
+    )
+    file_dem = global_data_dir / "dem" / gds.get_filename_dem(lat, lon)
+    files_era5 = list((global_data_dir / "era5").glob("*.nc"))
+    files_era5land = list((global_data_dir / "era5-land").glob("*.nc"))
+    files_lai = list((global_data_dir / "lai").glob("*.nc"))
+
+    data = {
+        "time": xr.DataArray(
+            pd.date_range(str(start_time), str(end_time), freq=TIMESTEP).rename("time")
+        )
+    }
+    era5_data = gds.extract_era5_data(
+        files_era5, files_era5land, lat, lon, start_time, end_time
+    )
+    data = {**data, **era5_data}
+
+    data["co2_conv"] = (
+        vc.co2_mass_fraction_to_kg_per_m3(
+            gds.extract_cams_data(files_cams, lat, lon, start_time, end_time)
+        )
+        * 1e6
+    )  # kg/m3 -> mg/m3
+
+    data["lai"] = vc.mask_data(
+        gds.extract_lai_data(files_lai, lat, lon, start_time, end_time), min_value=0.01
+    )
+
+    data["elevation"] = gds.extract_prism_dem_data(file_dem, lat, lon)
+
+    data["canopy_height"] = gds.extract_canopy_height_data(file_canopy_height, lat, lon)
+    data["reference_height"] = 0.7 * data["canopy_height"]
```

Comment:
Please add an explanation of why to choose `0.7` here.

Reply:
I raised a new issue about this in https://github.com/EcoExtreML/STEMMUS_SCOPE/issues/145